include/dxc/Test/HlslTestUtils.h

```diff
@@ -553,6 +563,27 @@ inline bool CompareDoubleULP(
   return AbsoluteDiff <= (uint64_t)ULPTolerance;
 }
 
+inline bool CompareDoubleEpsilon(
+    const double &Src, const double &Ref, float Epsilon,
+    hlsl::DXIL::Float32DenormMode Mode = hlsl::DXIL::Float32DenormMode::Any) {
+  if (Src== Ref) {
+    return true;
+  }
+  if (std::isnan(Src)) {
```

Comment:
What happens if Src is not nan and Ref is nan?

Reply:
Good question :). I coped the pattern from other functions here but I'm not sure what would happen. I'll debug that case to see and report back.

Reply:
In that case the comparison should fail at the end `(Src - NaN) = NaN`, `fabs(NaN) = NaN`, `NaN < Eps = false`.